compiler/parser/Makefile

```diff
@@ -36,20 +36,14 @@ include $(COMPILER_ROOT)/make/Makefile.compiler.subdirrules
 
 ifdef CHPL_DEVELOPER
 CLEAN_TARGS += \
-	*.tab.cpp \
-	*.yy.cpp \
-	*.tab.h \
-	*.output \
+               bison-chapel.output
 
 endif
 
-lex.yy.cpp: chapel.lex chapel.tab.h
-	flex -o$@ chapel.lex
-
-chapel.tab.h chapel.tab.cpp: chapel.ypp
-	bison --defines=chapel.tab.h -v -t $<
-	@if [ `grep "conflicts:" chapel.output | wc -l` -ne 0 ]; then echo "PROBLEM: chapel.y contains conflicts"; exit 1; fi;
-
+parser: FORCE
+	bison chapel.ypp
+	@if [ `grep "conflicts:" bison-chapel.output | wc -l` -ne 0 ]; then echo "PROBLEM: chapel.ypp contains conflicts"; exit 1; fi;
+	flex chapel.lex
```

Comment:
You are right that -o went away from flex.  That was naming the generated .cpp file and flex wasn’t
producing a header file.

Similarly the bison rule included —defines=chapel.tab.h which instructed bison to (a) generate the
optional header file (b) specify the name of that file.  Note that the bison command did not specify
the name of the generated .cpp file and so it defaulted to chapel.tab.cpp.



I wanted to get both files for each generator and I wanted to select the names.  This could be done
on the calls to bison/flex or it could be within the files themselves.


IMO it’s largely a coin flip which location is used.  If one relies on the Makefile then one feels more
confident that the makefile target matches what the processor will actually do.  But this is a false
security as the generated names for the header files are still embedded in the source code.

By moving all of the “command line” options in to the flex/bison files, a developer has the luxury
of saying “bison chapel.ypp” and getting the right output without referring to the Makefile to get
all of the options right.


Mike



On Feb 17, 2015, at 2:37 PM, Lydia Duncan <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/parser/Makefile<https://github.com/chapel-lang/chapel/pull/1331#discussion_r24861784>:

>
>  endif
>
> -lex.yy.cpp: chapel.lex chapel.tab.h
> -     flex -o$@ chapel.lex
> -
> -chapel.tab.h chapel.tab.cpp: chapel.ypp
> -     bison --defines=chapel.tab.h -v -t $<
> -     @if [ `grep "conflicts:" chapel.output | wc -l` -ne 0 ]; then echo "PROBLEM: chapel.y contains conflicts"; exit 1; fi;
> -
> +parser: FORCE
> +     bison chapel.ypp
> +     @if [ `grep "conflicts:" bison-chapel.output | wc -l` -ne 0 ]; then echo "PROBLEM: chapel.ypp contains conflicts"; exit 1; fi;
> +     flex chapel.lex


It looks like the -o portion of the flex line goes away. I know it was referring to an aspect of the make command, is there any functionality lost by doing this?

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/1331/files#r24861784>.